spring-mvc/src/main/java/com/kodemon/springmvc/controllers/UserController.java

```diff
@@ -110,4 +114,84 @@ public String login(@RequestParam String username, @RequestParam String password
             return "/login";
         }
     }
+
+    /**
+     * Block specified user. Only administrator can block other user and even admin cannot block another admin.
+     * User remain blocked until admin unblocks him.
+     *
+     * @param username Username of user to be blocked
+     * @param r servlet request to get logged user
+     * @param model Data to display
+     * @return JSP page name
+     */
+    @RequestMapping(value = "/blockUser", method = RequestMethod.POST)
+    public String blockUser(@RequestParam String username, ServletRequest r, Model model) {
+        HttpServletRequest request = (HttpServletRequest) r;
+        HttpSession session = request.getSession();
+        UserDTO toBeBlocked = userFacade.findUserByUserName(username).iterator().next();
+        if ((UserDTO)session.getAttribute("authenticatedUser") == null || !(((UserDTO)session.getAttribute("authenticatedUser")).isAdmin())) {
+            model.addAttribute("alert_warning", "You do not have permissions to do this");
+            LOG.error("User is not admin.");
+            model.addAttribute("trainer", toBeBlocked);
+            return "/user/detail";
+        }
+        if(toBeBlocked.isAdmin()) {
+            model.addAttribute("alert_warning", "Admin can not block another admin");
+            LOG.error("Admin trying to block admin");
+            model.addAttribute("trainer", toBeBlocked);
+            return "/user/detail";
+        }
+        LOG.info("User blocked");
+        userFacade.blockUser(toBeBlocked.getId());
+        model.addAttribute("alert_success", "User successfully blocked");
+        model.addAttribute("trainer", toBeBlocked);
+        return "/user/detail";
+    }
+
+    /**
+     * Unblock specified user if he is blocked. Only administrator can unblock user.
+     *
+     * @param username Username of user to be unblocked
+     * @param r servlet request to get logged user
+     * @param model Data to display
+     * @return JSP page name
+     */
+    @RequestMapping(value = "/unblockUser", method = RequestMethod.POST)
+    public String unblockUser(@RequestParam String username, ServletRequest r, Model model) {
+        HttpServletRequest request = (HttpServletRequest) r;
```

Comment:
Same as above. Return statement and the line just above it are same in both cases.